store/store.go

```diff
@@ -321,70 +322,42 @@ func (s *SnapUbuntuStoreRepository) getPurchasesFromURL(url *url.URL, channel st
 	return purchases, nil
 }
 
-func setMustBuy(snaps []*snap.Info) {
-	for _, info := range snaps {
-		if len(info.Prices) != 0 {
-			info.MustBuy = true
-		}
-	}
-}
-
-func hasPriced(snaps []*snap.Info) bool {
-	// Search through the list of snaps to see if any are priced
-	for _, info := range snaps {
-		if len(info.Prices) != 0 {
-			return true
-		}
-	}
-	return false
-}
-
 // decorateAllPurchases sets the MustBuy property of each snap in the given list according to the user's known purchases.
 func (s *SnapUbuntuStoreRepository) decoratePurchases(snaps []*snap.Info, channel string, auther Authenticator) error {
-	// Mark every non-free snap as must buy until we know better.
-	setMustBuy(snaps)
-
-	if auther == nil {
-		return nil
-	}
-
-	if !hasPriced(snaps) {
-		return nil
-	}
-
+	purchasesByID := make(map[string][]*purchase)
 	var err error
-	var purchasesURL *url.URL
 
-	if len(snaps) == 1 {
-		// If we only have a single snap, we should only find the purchases for that snap
-		purchasesURL, err = s.purchasesURI.Parse(snaps[0].SnapID + "/")
-		if err != nil {
-			return err
+	if auther != nil && s.purchasesURI != nil {
```

Comment:
that's fair. I think I added it as a knee-jerk, because if it's nil this kills the server, but given that it should be set by store's defaultconfig maybe the panic is ok.